src/dut.rs

```diff
@@ -747,6 +747,32 @@ impl SshInfo {
         // Return "x86_64" or "arm64"
         self.run_cmd_stdio("uname -m | sed s/aarch64/arm64/")
     }
+    pub fn get_rootdev(&self) -> Result<String> {
+        self.run_cmd_stdio("rootdev -s")
+    }
+    pub fn get_rootdisk(&self) -> Result<String> {
+        let rootdev = self.get_rootdev()?;
+        Ok(rootdev
+            .trim_end_matches(char::is_numeric)
+            .trim_end_matches('p')
+            .to_string())
+    }
+    pub fn get_partnum_root_a(&self) -> Result<String> {
+        let cmd_str = "source /usr/sbin/write_gpt.sh; load_base_vars; echo ${PARTITION_NUM_ROOT_A}";
+        self.run_cmd_stdio(cmd_str)
+    }
+    pub fn get_partnum_root_b(&self) -> Result<String> {
+        let cmd_str = "source /usr/sbin/write_gpt.sh; load_base_vars; echo ${PARTITION_NUM_ROOT_B}";
+        self.run_cmd_stdio(cmd_str)
+    }
+    pub fn get_partnum_kern_a(&self) -> Result<String> {
+        let cmd_str = "source /usr/sbin/write_gpt.sh; load_base_vars; echo ${PARTITION_NUM_KERN_A}";
+        self.run_cmd_stdio(cmd_str)
+    }
+    pub fn get_partnum_kern_b(&self) -> Result<String> {
+        let cmd_str = "source /usr/sbin/write_gpt.sh; load_base_vars; echo ${PARTITION_NUM_KERN_B}";
+        self.run_cmd_stdio(cmd_str)
+    }
```

Comment:
For `get_partnum_*`, How about just getting load_base_vars variables at once, not by each items...?

Reply:
combined them